chainerx_cc/chainerx/cuda/cuda_device/batch_norm.cc

```diff
@@ -312,10 +309,9 @@ Array CudaDevice::FixedBatchNorm(
         CHAINERX_ASSERT(&x.device() == &mean.device());
         CHAINERX_ASSERT(&x.device() == &var.device());
 
-        CHAINERX_ASSERT(x.dtype() == gamma.dtype());
-        CHAINERX_ASSERT(x.dtype() == beta.dtype());
-        CHAINERX_ASSERT(x.dtype() == mean.dtype());
-        CHAINERX_ASSERT(x.dtype() == var.dtype());
+        CHAINERX_ASSERT(gamma.dtype() == beta.dtype());
+        CHAINERX_ASSERT(gamma.dtype() == mean.dtype());
+        CHAINERX_ASSERT(gamma.dtype() == var.dtype());
```

Comment:
I think this can be relaxed to check the kind as in BatchNorm https://github.com/chainer/chainer/pull/6649/files#diff-aad3fa437c706c597c3720da0a68f7a0R139 It is cast anyway and it's technically supported?